lib/mutant/matcher/method.rb

```diff
@@ -1,133 +1,154 @@
 module Mutant
   class Matcher
-    # Matcher for subjects that are a specific method
+    # Abstract base class for method matchers
     class Method < self
-      include Adamantium::Flat, Concord::Public.new(:env, :scope, :target_method)
-      include AST::NodePredicates
+      include AbstractType,
+              Adamantium::Flat,
+              Concord::Public.new(:scope, :target_method, :evaluator)
 
       # Methods within rbx kernel directory are precompiled and their source
       # cannot be accessed via reading source location. Same for methods created by eval.
       BLACKLIST = %r{\A(kernel/|\(eval\)\z)}.freeze
 
-      # Enumerate matches
+      SOURCE_LOCATION_WARNING =
+        '%s does not have valid source location, unable to emit subject'.freeze
+
+      CLOSURE_WARNING =
+         '%s is dynamically defined in a closure, unable to emit subject'.freeze
```

Comment:
I think all warnings should be short and technically accurate. Single lines. BUT link to an exanation arthicle in the wiki / Readme section etc.

Mostly for the reason I do not like long warnings in the code, because there they are buried. The long explanation texts you where asking for are actually part of the missing documentation.